### PR TITLE
redesign health check to be generic

### PR DESCRIPTION
**Pull request by mhewedy**

### CODE REVIEW

File: ip/ip.go [hunk 1]
```diff
@@ -17,26 +15,11 @@ type addr struct {
 
 func GetIpAddress(vmName string) (string, error) {
 
-	health, err := hypervisor.HealthCheck(vmName)
+	err := hypervisor.HealthCheck(vmName)
```
Comment (akhiljns) [inline ip/ip.go:18]: status should also be checked here

```
status, err := hypervisor.HealthCheck(vmName)
if err != nil {
    return "", err
}

if !status {
    return "", fmt.Errorf("VM health check failed")
}
```

Comment (akhiljns) [reply ip/ip.go:18]: everything else looks good!

Comment (mhewedy) [reply ip/ip.go:18]: hypervisor.HealthCheck utility function only returns as error. 

so the HealthCheck method of the type Hypervisor return (bool, error)
and the HealthCheck function returns error only.

Comment (akhiljns) [reply ip/ip.go:18]: ahh right cool, merging this then

**Pull request merged**